backend/src/plugins/kube.ts

```diff
@@ -93,3 +108,34 @@ const getCurrentNamespace = async () => {
     }
   });
 };
+
+const getSAToken = async () => {
+  return new Promise<string>((resolve, reject) => {
+    fs.readFile('/var/run/secrets/kubernetes.io/serviceaccount/token', (err, data) => {
+      if (err) {
+        reject(err);
+      }
+      resolve(String(data));
+    });
+  });
+};
+
+const getSATokenLocal = async () => {
+  const promSA = await coreV1Api
+    .readNamespacedServiceAccount('prometheus-k8s', 'openshift-monitoring')
+    .then((res) => res.body as V1ServiceAccount);
+  let promTokenName = '';
+  for (let i = 0; i < promSA.secrets.length; i++) {
+    if (promSA.secrets[i].name.includes('token')) {
+      promTokenName = promSA.secrets[i].name;
+    }
+  }
```

Comment:
- What happens if you don't the service account?
- What happens if you don't find one that matches?
- What happens if you find one early?
    - Do you not want to use `.find`?

You make a lot of assumptions about the state of things, and I think you need to handle what-ifs -- what if you don't get it back, what if your list is empty.